frontend/ui/widget/qrwidget.lua

```diff
@@ -4,34 +4,43 @@ QRWidget shows a QR code for a given text.
 
 local ImageWidget = require("ui/widget/imagewidget")
 local RenderImage = require("ui/renderimage")
-local Widget = require("ui/widget/widget")
 local logger = require("logger")
 local qrencode = require("qrencode")
 
-local QRWidget = Widget:new{
+local QRWidget = ImageWidget:extend{
     scale_factor = 0,
     text = ""
     -- see ImageWidget for other options.
 }
 
 function QRWidget:init()
-    local ok, ret = qrencode.qrcode(self.text)
+    local ok, ret = qrencode.qrcode(self.text:sub(1, 2953))
```

Comment:
I would check if it overflows, and if yes, truncate a bit less to have room to add `[...truncated...]` so the receiver know it did not get the full intended text.